tests/metrics/grant_availability.py

```diff
@@ -0,0 +1,187 @@
+                         bob=make_random_bob(),
+                         label=label,
+                         rate=RATE)
+    return policy
+
+
+def collect(alice: Alice,
+            handpicked_ursulas: Optional[Set[Ursula]] = None,
+            iterations: Optional[int] = None,
+            run_forever: bool = False
+            ) -> None:
+    """Collects grant success and failure rates."""
+    policies, i, success, fail = dict(), 0, 0, 0
+    while True:
+        print(f'Attempt {i+1} of {iterations}')
+        start = maya.now()
+        try:
+            policy = metric_grant(alice=alice, handpicked_ursulas=handpicked_ursulas)
+        except Exception as e:
+            fail += 1
+            print(f'GRANT FAIL\n{e}')
+        else:
+            success += 1
+            policies[policy.public_key.hex()] = policy  # track
+            print(f"PEK:{policy.public_key.hex()}")
+
+        # timeit
+        end = maya.now()
+        delta = end - start
+        print(f"Completed in {(delta).total_seconds()} seconds.")
+
+        if i == iterations and not run_forever:
+            break  # Exit
+
+        # score
+        elif i != iterations:
+            if fail > 0:
+                print(f'{fail}/{i+1} ({(fail/(i+1))*100}%) failure rate')
+            if success > 0:
+                print(f'{success}/{i+1} ({(success/(i+1))*100}%) success rate')
+            print(f'Waiting {SAMPLE_RATE} seconds until next sample. ')
+            time.sleep(SAMPLE_RATE)
+        i += 1
+
+
+def make_alice(known_nodes: Optional[Set[Ursula]] = None):
+    """Initializes a new 'persistent alice configuration' for grant metrics collection."""
+    alice_config = AliceConfiguration(
+        provider_uri=ETHEREUM_PROVIDER_URI,
+        checksum_address=ALICE_ADDRESS,
+        signer_uri=SIGNER_URI,
+        config_root=os.path.join(TEMP_ALICE_DIR),
+        domains={DOMAIN},
+        known_nodes=known_nodes,
+        start_learning_now=False,
+        federated_only=False,
+        learn_on_same_thread=True,
+        gas_strategy=GAS_STRATEGY
+    )
+
+    alice_config.initialize(password=INSECURE_PASSWORD)
+    alice_config.keyring.unlock(password=INSECURE_PASSWORD)
+    alice = alice_config.produce(client_password=SIGNER_PASSWORD)
+    alice.start_learning_loop(now=True)
+    return alice
+
+
+def setup():
+    """Prepares the filesystem and logger for grsnt metrics collection"""
+    shutil.rmtree(TEMP_ALICE_DIR, ignore_errors=True)
+    GlobalLoggerSettings.start_console_logging()
+    GlobalLoggerSettings.set_log_level('info')
+
+
+def aggregate_nodes() -> Tuple[Set[Ursula], Set[Ursula]]:
+    """generates ursulas from URIs used in grant metrics collection"""
+
+    seednodes = set()
+    if DEFAULT_SEEDNODE_URIS:
+        for uri in DEFAULT_SEEDNODE_URIS:
+            ursula = Ursula.from_seed_and_stake_info(seed_uri=uri, federated_only=False)
+            seednodes.add(ursula)
+
+    handpicked_ursulas = set()
```

Comment:
This helps developers/Alices discover the availability of a subset of Ursulas that they specify in a URI list?

Reply:
@arjunhassard - Yes, exactly.